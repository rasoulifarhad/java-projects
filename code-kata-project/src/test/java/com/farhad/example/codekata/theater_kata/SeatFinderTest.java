package com.farhad.example.codekata.theater_kata;

import static com.farhad.example.codekata.theater_kata.ModelObjects.standardTheater2;
import static java.util.stream.Collectors.toList;

import java.util.List;
import java.util.stream.Stream;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

public class SeatFinderTest {

    @Test
    public void suggestSeatsInMiddle() {

        Theater theater = standardTheater2();
        List<Seat> booked = 
            Stream.of("A2", "A3", "A4", "A5", "A6")
                .map(Seat::new)
                .collect(toList());

        SeatFinder seatFinder = new SeatFinder(theater, booked);
        List<Seat> suggestedSeats = seatFinder.suggest(2);
        Approvals.verifyAll("Suggested seats", suggestedSeats);
    }
}
