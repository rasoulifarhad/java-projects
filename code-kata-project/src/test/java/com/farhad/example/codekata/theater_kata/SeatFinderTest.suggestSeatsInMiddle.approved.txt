Suggested Seats[0] = B4
Suggested Seats[1] = B5